ui/nuxeo-page-provider-display-behavior.js

```diff
@@ -447,27 +445,27 @@ export const PageProviderDisplayBehavior = [
       }
     },
 
-    _observeEmptyLabel() {
-      this._computedEmptyLabel = this.emptyLabel;
+    _computeLabel() {
+      let label = this.emptyLabel ? this.emptyLabel : this.i18n('label.noResults');
+      if (this.loading) {
+        label = this.i18n('label.loading');
+      } else if (this.filters && this.filters.length > 0) {
+        label = this.emptyLabelWhenFiltered ? this.emptyLabelWhenFiltered : this.i18n('label.noResultsWhenFiltered');
+      }
+      return label;
     },
 
     _quickFilterChanged() {
       this.fetch();
     },
 
     _updateFlags() {
-      if (!this.loading) {
-        this.size = this.items && this.items.length > 0 ? this.items.length : 0;
-        this._isSelectAllIndeterminate =
-          this.selectedItems && this.selectedItems.length > 0 && this.selectedItems.length < this.size;
-        this._isSelectAllChecked =
-          this.selectedItems && this.selectedItems.length > 0 && this.selectedItems.length === this.size;
-        this._isEmpty = this.size === 0;
-        this._computedEmptyLabel =
-          this.filters && this.filters.length > 0 ? this.emptyLabelWhenFiltered : this.emptyLabel;
-      } else {
-        this._computedEmptyLabel = this.i18n('label.loading');
-      }
+      this.size = this.items && this.items.length > 0 ? this.items.length : 0;
+      this._isSelectAllIndeterminate =
+        this.selectedItems && this.selectedItems.length > 0 && this.selectedItems.length < this.size;
+      this._isSelectAllChecked =
+        this.selectedItems && this.selectedItems.length > 0 && this.selectedItems.length === this.size;
+      this._isEmpty = this.size === 0;
```

Comment:
Maybe it's worth extracting the selected items size:
```
this.size = this.items && this.items.length > 0 ? this.items.length : 0;
const selectedItemsSize = this.selectedItems && this.selectedItems.length > 0 ? this.selectedItems.length : 0;
this._isSelectAllIndeterminate = selectedItemsSize < this.size;
this._isSelectAllChecked = selectedItemsSize === this.size;
this._isEmpty = this.size === 0;
```
Also, you can probably just get those lengths using `Array.isArray()`:
```
this.size = Array.isArray(this.items) ? this.items.length : 0;
const selectedItemsSize = Array.isArray(this.selectedItems) ? this.selectedItems.length : 0;
```
(should be `null`/`undefined` safe)

Reply:
Thanks!